src/components/forms/ComboBox/ComboBox.tsx

```diff
@@ -0,0 +1,284 @@
+      if (newIndex < 0) {
+        setIsOpen(false)
+        setFocusedValue(undefined)
+        setFocusMode(FocusMode.Input)
+      } else {
+        // eslint-disable-next-line security/detect-object-injection
+        const newOption = filteredOptions[newIndex]
+        setFocusedValue(newOption.value)
+      }
+    }
+  }
+
+  const handleInputKeyDown = (event: KeyboardEvent): void => {
+    if (event.key == 'Escape') {
+      setIsOpen(false)
+    } else if (event.key == 'ArrowDown' || event.key == 'Down') {
+      event.preventDefault()
+      setIsOpen(true)
+      changeFocus(Direction.Down)
+      setFocusMode(FocusMode.Item)
+    }
+  }
+
+  const handleInputClick = (): void => {
+    setFocusMode(FocusMode.Input)
+    setIsOpen(true)
+  }
+
+  const handleInputBlur = (event: FocusEvent<HTMLInputElement>): void => {
+    const target = event.relatedTarget
+
+    if (
+      !target ||
+      (target instanceof Node && !containerRef.current?.contains(event.target))
+    ) {
+      setFocusMode(FocusMode.None)
+      setIsOpen(false)
+    }
+  }
+
+  const handleListItemKeyDown = (event: KeyboardEvent): void => {
+    if (event.key == 'Escape') {
+      setIsOpen(false)
+      if (!selectedValue) {
+        setInputValue('')
+      }
+      setFocusedValue(undefined)
+      setFocusMode(FocusMode.Input)
+    } else if (event.key == 'ArrowDown' || event.key == 'Down') {
+      event.preventDefault()
+      changeFocus(Direction.Down)
+    } else if (event.key == 'ArrowUp' || event.key == 'Up') {
+      event.preventDefault()
+      changeFocus(Direction.Up)
+    }
+  }
+
+  const handleListItemClick = (option: ComboBoxOption): void => {
+    setSelectedValue(option.value)
+    setInputValue(option.label || option.value)
+    setIsOpen(false)
+    setFocusMode(FocusMode.Input)
+  }
+
+  const handleListItemMouseMove = (option: ComboBoxOption): void => {
+    if (focusedValue !== option.value) {
+      setFocusedValue(option.value)
+      setFocusMode(FocusMode.Item)
+    }
+  }
+
+  const handleArrowClick = (): void => {
+    if (isOpen) {
+      setFocusMode(FocusMode.Input)
+      setIsOpen(false)
+    } else {
+      if (selectedValue) {
+        setFocusMode(FocusMode.Item)
+      }
+      setIsOpen(true)
+    }
+  }
+
+  const handleClearClick = (): void => {
+    setInputValue('')
+    setSelectedValue(undefined)
+    setFocusMode(FocusMode.Input)
+  }
+
+  const containerClasses = classnames('usa-combo-box', className, {
+    'usa-combo-box--pristine': selectedValue,
+  })
+
+  return (
+    <div
+      data-testid="combo-box"
+      className={containerClasses}
+      id={id}
+      ref={containerRef}>
+      <select
+        className="usa-select usa-sr-only usa-combo-box__select"
+        name={name}
+        aria-hidden
+        tabIndex={-1}
+        value={selectedValue}
+        {...selectProps}>
+        {options.map((option) => (
+          <option key={option.value} value={option.value}>
```

Comment:
The consumer has to make sure their option values are unique. Maybe we should rename value to`id` in `ComboBoxOption` to suggest this more in the naming.